pkg/compact/compact.go

```diff
@@ -724,40 +724,49 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 
 	// Once we have a plan we need to download the actual data.
 	begin := time.Now()
+	toCompactDirs := make([]string, len(toCompact))
 
-	toCompactDirs := make([]string, 0, len(toCompact))
-	for _, meta := range toCompact {
+	var eg errgroup.Group
+	for i, meta := range toCompact {
```

Comment:
If we use the more idiomatic slice generation, then we can revert those change as well
```suggestion
	for _, meta := range toCompact {
```